### PR TITLE
move to vitest

### PR DESCRIPTION
**Pull request by hans-lizihan**
vitest requires node version 18. so updated workflow.yaml as well.

### CODE REVIEW

File: lib/people.js [hunk 1]
```diff
@@ -191,11 +191,11 @@ class MixpanelPeople extends ProfileHelpers() {
     */
     track_charge(distinct_id, amount, properties, modifiers, callback) {
         if (typeof(properties) === 'function' || !properties) {
-            callback = properties || function() {};
+            callback = properties || undefined;
```
Comment (hans-lizihan) [inline lib/people.js:194]: make sure the callback is conditionally passed in `send_request`

File: test/alias.js [hunk 2]
```diff
@@ -1,41 +1,37 @@
...
-        next();
-    },
-
-    "calls send_request with correct endpoint and data": function(test) {
+const Mixpanel = require('../lib/mixpanel-node');
+
+describe('alias', () => {
+    let mixpanel;
+    let sendRequestMock;
+    beforeAll(() => {
```
Comment (tdumitrescu) [inline test/alias.js:6]:
```suggestion
    beforeEach(() => {
```

File: test/config.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-        cb();
-    },
-
-    "is set to correct defaults": function(test) {
-        test.deepEqual(this.mixpanel.config, {
+const Mixpanel = require('../lib/mixpanel-node');
+
+describe('config', () => {
+    let mixpanel;
+    beforeAll(() => {
```
Comment (tdumitrescu) [inline test/config.js:5]: I would also make this one a `beforeEach` - make sure each test case has a fresh mp instance

File: test/import.js [hunk 4]
```diff
@@ -1,369 +1,306 @@
...
-    tearDown: function(next) {
-        this.mixpanel.send_request.restore();
-        this.clock.restore();
-
-        next();
-    },
-
-    "calls send_request with correct endpoint and data": function(test) {
-        var event = "test",
+six_days_ago_timestamp = mock_now_time - 1000 * 60 * 60 * 24 * 6;
```
Comment (tdumitrescu) [inline test/import.js:7]:
```suggestion
    six_days_ago_timestamp = mock_now_time - 1000 * 60 * 60 * 24 * 6;
```

File: test/import.js [hunk 5]
```diff
@@ -1,369 +1,306 @@
...
+
+describe('import', () => {
+    let mixpanel;
+    beforeAll(() => {
```
Comment (tdumitrescu) [inline test/import.js:11]:
```suggestion
    beforeEach(() => {
```

File: test/import.js [hunk 6]
```diff
@@ -1,369 +1,306 @@
...
+        mixpanel = Mixpanel.init('token', { secret: 'my api secret' });
+
+        vi.spyOn(mixpanel, 'send_request');
+    }),
+    afterAll(() => {
```
Comment (tdumitrescu) [inline test/import.js:16]: return this func from the before-func, for consistency with your other tests

File: test/import.js [hunk 7]
```diff
@@ -1,369 +1,306 @@
...
 
-    tearDown: function(next) {
-        this.mixpanel.send_request.restore();
-        this.clock.restore();
+        vi.spyOn(mixpanel, 'send_request');
+    });
 
-        next();
-    },
+    afterEach(() => {
```
Comment (tdumitrescu) [inline test/import.js:138]: return from before-func

File: test/people.js [hunk 8]
```diff
@@ -1,259 +1,242 @@
...
+    const endpoint = '/engage';
+    const distinct_id = 'user1';
+    const token = 'token';
+    let send_request;
+    let mixpanel;
+    beforeEach(() => {
+        send_request = vi.fn();
+
+        mixpanel = Mixpanel.init(token);
+        mixpanel.send_request = send_request
```
Comment (tdumitrescu) [inline test/people.js:14]: Seems like you're inconsistent between using this approach and `vi.spyOn(mixpanel, 'send_request')`. I would think spyOn is preferable, but is there some context I'm missing?

File: test/send_request.js [hunk 9]
```diff
@@ -1,46 +1,43 @@
...
 let HttpsProxyAgent;
 
-exports.send_request = {
-    setUp: function(next) {
-        HttpsProxyAgent = Sinon.stub();
+describe("send_request", () => {
+    let mixpanel;
+    let http_emitter;
+    let res;
+    beforeEach((next) => {
```
Comment (tdumitrescu) [inline test/send_request.js:13]:
```suggestion
    beforeEach(() => {
```

File: test/track.js [hunk 10]
```diff
@@ -1,376 +1,333 @@
...
-        this.clock.restore();
-
-        next();
-    },
-
-    "calls send_request with correct endpoint and data": function(test) {
-        var event = "test",
+describe('track', () => {
+    let mixpanel;
+    beforeAll(() => {
```
Comment (tdumitrescu) [inline test/track.js:12]:
```suggestion
    beforeEach(() => {
```

**Review by tdumitrescu — COMMENTED** [review]
Awesome. Surprised to see coverage is already so good. Just a bit of cleanup suggested but largely good to go

File: test/people.js (hunk 8 above)
Comment (hans-lizihan) [reply test/people.js:14]: good catch.. i think when i did the conversion, this was the way sinon stubs this function.

**Comment by hans-lizihan** [comment]
@tdumitrescu updated

**Review by tdumitrescu — APPROVED** [review]
🎉

**Pull request merged**